### ISSUE TITLE
Filtering + Paging

### ISSUE THREAD

**Issue by thnk2wn** [Knockout-Contrib/KoGrid#53]
How does filtering work with paging? If I go to http://ericmbarnard.github.com/KoGrid/examples/Complex-Server-Side-Paging.html and navigate to page 2, I can see that NIKE is there but filtering on it doesn't work. It will on page 1.

**Comment by cmolendijk** [comment]
I've got the same Issue. The filter is working but the page is not going back to page 1 to show the filtered result.

By changing the getPagedDataAsync function to the following i got the KoGrid to show the result:

```
        this.getPagedDataAsync = function (pageSize, page, filterInfo, sortInfo) {
            setTimeout(function () {

                var filteredData = filter(ko.observableArray(items)(), filterInfo);
                var sortedData = sort(filteredData, sortInfo);
                if (sortedData.length < pageSize) {
                    page = 1;
                    self.currentPage = ko.observable(1);
                }
                var pagedData = sortedData.slice((page - 1) * pageSize, page * pageSize);

                self.addressArray(pagedData);

            }, 0);
        };
```

But the numbers in the paging aren't changing. Does someone know a solution for this?

**Comment by orneryd** [comment]
fixed by forcing the currentpage back to page 1 whenever the filter changes

in commit: https://github.com/Crash8308/sKoGrid/commit/f578a3e59beafa62ba1c1fd71962c7561bdd7b6b

https://github.com/Crash8308/sKoGrid

**Comment by ericmbarnard** [comment]
Closing as this will be fixed in PR #67
